gulp/modules/Utilities.js

```diff
@@ -0,0 +1,148 @@
+var fs = require('fs');
+var Config = require('./Config');
+var path = require('path');
+
+/**
+ * Helpful assortment of Utilites used throughout the build
+ */
+var Utilities = function() {
+	/**
+     * Retrieves all months
+     * @returns {array} An array with the month names as strings
+     */
+	this.getMonths = function() {
+		return [
+            "January", "February", "March",
+            "April", "May", "June", "July",
+            "August", "September", "October",
+            "November", "December"
+		];
+	};
+	/**
+     * Process and return the list of files for the component either by whats in the directory or what's listed in the manifest
+     * @param {array} fileArray     An array with just file names
+     * @param {string} folderPath   A string containing the path to where the files are located.
+     * @returns {array}              Returns an array containing new list and order for files  
+     */
+	this.getManifestFileList = function(fileArray, folderPath) {
+		var newArray = [];
+		
+		if(typeof fileArray != "undefined" || fileArray != undefined) {
+			newArray = fileArray.map(function(file, i) {
+				return folderPath + '/' + file;
+			});
+			return newArray;
+		} else {
+			newArray.push(folderPath + '/*.html');
+			return newArray;
+		}
+	};
+	/**
+     * Parse a specific component's manifest (JSON) file
+     * @param {string}  jsonFileLocation string containing the path to folder/file of the JSON manifest
+     * @returns {object} Object Containing the manifest data in an object 
+     */
+	this.parseManifest = function(jsonFileLocation) {
+        var manifest = fs.readFileSync(jsonFileLocation);
+		return JSON.parse(manifest.toString());
+	};
+	/**
+     * Return an array of folders from specified directory without files
+     * @param {string} dir  Path to directory
+     * @returns {array}      Returns an array of folders within an folder wihout files
+     */
+	this.getFolders = function(dir) {
+        try {
```

Comment:
Same comment as above. Tabs vs spaces again. Setting your IDE to use spaces instead of tabs might fix this in the future.